torch/distributed/_sharded_tensor/ops/equals.py

```diff
@@ -0,0 +1,51 @@
+import torch
+import torch.distributed as dist
+
+def _communicate_result(result, pg):
+    # Gather results from all ranks.
+    if result:
+        result_tensor = torch.ones(1, device=torch.cuda.current_device())
+    else:
+        result_tensor = torch.zeros(1, device=torch.cuda.current_device())
+
+    dist.all_reduce(result_tensor, group=pg)
+
+    expected_result = torch.ones(1, device=torch.cuda.current_device()) * dist.get_world_size(pg)
+
+    return torch.equal(result_tensor, expected_result)
+
+def equal(types, args=(), kwargs=None):
+    from torch.distributed._sharded_tensor import ShardedTensor
+
+    if len(args) != 2:
+        raise ValueError('Expected two arguments for torch.equal')
+
+    result = True
+    st1 = args[0]
+    st2 = args[1]
+    if not(isinstance(st1, ShardedTensor) and isinstance(st2, ShardedTensor)):
+        raise TypeError('Both arguments to torch.equal need to be of type ShardedTensor')
```

Comment:
nit: maybe print out the type of st1 and st2 in case they are not sharded tensors